### PR TITLE
_9base: init at unstable-2019-09-11

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Description of changes

The configuration is a bit of a pain and I've not really looked at the concequenses of putting troff stuff in a different output (since it's pretty huge), also I don't like this "/yacc" dir. but `rc` & `ls` works well
Thoughts welcome around those issues mentioned

```sh
λ nix bundle --bundler github:NixOS/bundlers#toDockerImage .#pkgsStatic._9base

λ docker load < 9base-static-x86_64-unknown-linux-musl-unstable-2019-09-11.tar.gz
41bf90e092c2: Loading layer [==================================================>]  10.24kB/10.24kB
7a88d4804328: Loading layer [==================================================>]  6.093MB/6.093MB
78a2a254bd17: Loading layer [==================================================>]  102.4kB/102.4kB
Loaded image: 9base-static-x86_64-unknown-linux-musl-unstable-2019-09-11:latest

λ docker run -it 9base-static-x86_64-unknown-linux-musl-unstable-2019-09-11 rc
; ls
.dockerenv
bin
dev
etc
lib
nix
nix-support
proc
sys
yacc
;
```

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [X] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 214932` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>_9base</li>
  </ul>
</details>

**Comment by 06kellyjac** [comment]
aarch64-darwin will need some porting work that I'm not prepared for so I've marked it as broken for now

**Comment by Aleksanaa** [comment]
What's blocking this pr from being merged?

**Review by 9glenda — APPROVED** [review]

**Review by 9glenda — APPROVED** [review]

**Comment by 06kellyjac** [comment]
Nothing is blocking it from being merged, it needed an approval but 9glenda provided one. Just need someone to click the button
I could rebase it but since there's no conflicts I'd say there's no need

---

aarch64-darwin support would be nice but not a hard requirement & ideally would get some upstream help
I tried asking in the suckless IRC a long while ago but no reply

File: pkgs/tools/system/9base/default.nix
Comment (Artturin) [inline pkgs/tools/system/9base/default.nix:38]:
```suggestion
  lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

**Review by Artturin — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/9base/config-substitutions.patch
Comment (06kellyjac) [reply pkgs/tools/system/9base/config-substitutions.patch:15]: That's because of `./getcallerpc-use-macro-or-stub.patch` making it unused. I've modified the patch to just strip that

File: pkgs/tools/system/9base/default.nix
Comment (06kellyjac) [reply pkgs/tools/system/9base/default.nix:38]: I think this suggestion is missing a `!`, added with that amendment

File: pkgs/tools/system/9base/default.nix
Comment (06kellyjac) [reply pkgs/tools/system/9base/default.nix:36]: done

File: pkgs/tools/system/9base/default.nix
Comment (06kellyjac) [reply pkgs/tools/system/9base/default.nix]: This is a pretty common convention among suckless tools in nixpkgs as they're expected to be patched so I'd rather leave it but if you insist I'll drop it

File: pkgs/tools/system/9base/default.nix
Comment (06kellyjac) [reply pkgs/tools/system/9base/default.nix]: I've taken this approach in a WIP commit, LMK what you think and I'll squash it in

**Comment by 06kellyjac** [comment]
cross compile for ppc64 worked

```
shrinking /nix/store/b7nr14qf6dwd2ngfflkfkn8gi0fawf8n-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11/bin/strings
checking for references to /build/ in /nix/store/b7nr14qf6dwd2ngfflkfkn8gi0fawf8n-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11...
patching script interpreter paths in /nix/store/b7nr14qf6dwd2ngfflkfkn8gi0fawf8n-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11
stripping (with command powerpc64-unknown-linux-gnuabielfv2-strip and flags -S -p) in  /nix/store/b7nr14qf6dwd2ngfflkfkn8gi0fawf8n-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11/lib /nix/store/b7nr14qf6dwd2ngfflkfkn8gi0fawf8n-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11/bin
shrinking RPATHs of ELF executables and libraries in /nix/store/nzwgi0qlv3s7ki4h2d7fy5kxw6xfh42p-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-man
checking for references to /build/ in /nix/store/nzwgi0qlv3s7ki4h2d7fy5kxw6xfh42p-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-man...
gzipping man pages under /nix/store/nzwgi0qlv3s7ki4h2d7fy5kxw6xfh42p-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-man/share/man/
patching script interpreter paths in /nix/store/nzwgi0qlv3s7ki4h2d7fy5kxw6xfh42p-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-man
shrinking RPATHs of ELF executables and libraries in /nix/store/9s3kafk846nc9fzbv5r2c2chh1mql006-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-troff
checking for references to /build/ in /nix/store/9s3kafk846nc9fzbv5r2c2chh1mql006-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-troff...
patching script interpreter paths in /nix/store/9s3kafk846nc9fzbv5r2c2chh1mql006-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-troff
stripping (with command powerpc64-unknown-linux-gnuabielfv2-strip and flags -S -p) in  /nix/store/9s3kafk846nc9fzbv5r2c2chh1mql006-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11-troff/lib
/nix/store/b7nr14qf6dwd2ngfflkfkn8gi0fawf8n-9base-powerpc64-unknown-linux-gnuabielfv2-unstable-2019-09-11
$ file ./result/bin/rc
./result/bin/rc: ELF 64-bit MSB executable, 64-bit PowerPC or cisco 7500, OpenPOWER ELF V2 ABI, version 1 (SYSV), dynamically linked, interpreter /nix/store/lr0gzmkqqgjsw11kjsa7167rnvh54mqp-glibc-powerpc64-unknown-linux-gnuabielfv2-2.37-8/lib/ld64.so.2, for GNU/Linux 3.10.0, not stripped
```

**Comment by 06kellyjac** [comment]
If I could get another review that'd be fantastic @Artturin, I think everything is addressed, LMK

**Review by Artturin — APPROVED** [review]

**Comment by 06kellyjac** [comment]
Thanks

**Pull request merged**
